src/widgets/qownnotesmarkdowntextedit.cpp

```diff
@@ -308,6 +308,59 @@ QMargins QOwnNotesMarkdownTextEdit::viewportMargins() {
 #endif
 }
 
+void QOwnNotesMarkdownTextEdit::setText(const QString &text) {
+    QOwnNotesMarkdownHighlighter *h = dynamic_cast<QOwnNotesMarkdownHighlighter*>(_highlighter);
+
+    //check for comment block
+    if (!text.contains(QStringLiteral("<!--"))) {
+        h->setCommentHighlighting(false);
+    }
+
+    //check for code blocks
+    if (!text.contains(QStringLiteral("```"))) {
+        h->setCodeHighlighting(false);
+    }
+
+    //check for front matter
+    if (h->document()->firstBlock().text() != "---") {
+
```

Comment:
Did you miss a `setFrontmatterHighlighting`?

Reply:
ahh yes. Fixed